barretenberg/cpp/src/barretenberg/bb/main.cpp

```diff
@@ -164,10 +164,7 @@ bool proveAndVerifyHonkAcirFormat(acir_format::AcirFormat constraint_system, aci
     // Construct a bberg circuit from the acir representation
     auto builder = acir_format::create_circuit<Builder>(constraint_system, 0, witness);
 
-    // TODO(https://github.com/AztecProtocol/barretenberg/issues/811): Add a buffer to the expected circuit size to
-    // account for the addition of "gates to ensure nonzero polynomials" (in Honk only).
-    const size_t additional_gates_buffer = 15; // conservatively large to be safe
-    size_t srs_size = builder.get_circuit_subgroup_size(builder.get_total_circuit_size() + additional_gates_buffer);
+    size_t srs_size = builder.get_circuit_subgroup_size(builder.get_total_circuit_size());
```

Comment:
should this not just be builder.num_gates now? also add a comment explaining why we don't have to add any buffers/do any computations here anymore